From ee9962e95b81e547191f59051622198af369069f Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:51:18 -0400
Subject: [fs] nfsd: nfsd_vfs_read doesn't use file handle parameter

Message-id: <1403560314-32525-137-git-send-email-bfields@redhat.com>
Patchwork-id: 84507
O-Subject: [Patch RHEL7 136/172] nfsd4: nfsd_vfs_read doesn't use file handle parameter
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 02fe4707740e7c8a3d0ec34ffbf4630d7d41ca68
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/vfs.c b/fs/nfsd/vfs.c
index 57205f9..62a229a 100644
--- a/fs/nfsd/vfs.c
+++ b/fs/nfsd/vfs.c
@@ -972,7 +972,7 @@ static int nfsd_direct_splice_actor(struct pipe_inode_info *pipe,
 }
 
 static __be32
-nfsd_vfs_read(struct svc_rqst *rqstp, struct svc_fh *fhp, struct file *file,
+nfsd_vfs_read(struct svc_rqst *rqstp, struct file *file,
               loff_t offset, struct kvec *vec, int vlen, unsigned long *count)
 {
  mm_segment_t oldfs;
@@ -1132,7 +1132,7 @@ __be32 nfsd_read(struct svc_rqst *rqstp, struct svc_fh *fhp,
  if (ra && ra->p_set)
   file->f_ra = ra->p_ra;
 
- err = nfsd_vfs_read(rqstp, fhp, file, offset, vec, vlen, count);
+ err = nfsd_vfs_read(rqstp, file, offset, vec, vlen, count);
 
  /* Write back readahead params */
  if (ra) {
@@ -1161,7 +1161,7 @@ nfsd_read_file(struct svc_rqst *rqstp, struct svc_fh *fhp, struct file *file,
     NFSD_MAY_READ|NFSD_MAY_OWNER_OVERRIDE);
   if (err)
    goto out;
-  err = nfsd_vfs_read(rqstp, fhp, file, offset, vec, vlen, count);
+  err = nfsd_vfs_read(rqstp, file, offset, vec, vlen, count);
  } else /* Note file may still be NULL in NFSv4 special stateid case: */
   err = nfsd_read(rqstp, fhp, offset, vec, vlen, count);
 out:
-- 
1.7.1